src/languages/c.js

```diff
@@ -57,14 +57,14 @@ export default function(hljs) {
   const NUMBERS = {
     className: 'number',
     variants: [
-      { begin: '\\b(0b[01\']+)' },
-      { begin: '(-?)\\b([\\d\']+(\\.[\\d\']*)?|\\.[\\d\']+)((ll|LL|l|L)(u|U)?|(u|U)(ll|LL|l|L)?|f|F|b|B)' },
-      { begin: '(-?)(0[xX][a-fA-F0-9]+(?:\'[a-fA-F0-9]+)*(?:\\.[a-fA-F0-9]*(?:\'[a-fA-F0-9]*)*)?(?:[pP][-+]?[0-9]+)?|\\b\\d+(?:\'\\d+)*(?:\\.\\d*(?:\'\\d*)*)?(?:[eE][-+]?\\d+)?)' }
+      { begin: '\\b(0b[01\']+)' },  
+      { begin: '(-?)\\b([\\d\']+(\\.[\\d\']*)?|\\.[\\d\']+)((ll|LL|l|L)(u|U)?|(u|U)(ll|LL|l|L)?|f|F|b|B)' },  
+      { begin: '(-?)(0[xX][a-fA-F0-9]+(?:\'[a-fA-F0-9]+)*(?:\\.[a-fA-F0-9]*(?:\'[a-fA-F0-9]*)*)?(?:[pP][-+]?[0-9]+)?)' },  
+      { begin: '(-?)\\b\\d+(?:\'\\d+)*(?:\\.\\d*(?:\'\\d*)*)?(?:[eE][-+]?\\d+)?' }  
```

Comment:
Now if we could please just switch to `match` vs `begin` and literal regex: `/.../` (vs strings)... then this is good to go I think. I'm also happy to do this cleanup myself, just might be a little while.

Reply:
@joshgoebel i've changed the match to begin , i also managed to correct for a issue in detectign hex longs that went previously undetected please review if any changes are needed
![before](https://github.com/user-attachments/assets/d567198d-4734-4e45-bb67-11a5385c5a2c)
![after](https://github.com/user-attachments/assets/1034397c-1d44-40a2-a1c0-9dfa40027952)